metis-indexing/src/main/java/eu/europeana/indexing/tiers/media/EmbeddableMedia.java

```diff
@@ -5,30 +5,54 @@
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.EnumSet;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 /**
  * This class determines whether a given web resource represents embeddable media.
  */
 final class EmbeddableMedia {
 
-  private static final Collection<String> PREFIXES = Arrays.asList(
-      "urn:soundcloud:"
-      , "http://player.vimeo.com/video/"
-      , "https://vimeo.com/"
-      , "https://sketchfab.com/models/"
-      , "http://sounds.bl.uk/embed/"
-      , "http://eusounds.ait.co.at/"
-      , "http://images3.noterik.com/edna/domain/euscreenxl/"
-
-      , "http://www.ccma.cat/tv3/alacarta/programa/titol/video/"
-      , "http://www.ina.fr/video/"
-      , "http://www.ina.fr/*/video/"
-      , "http://api.picturepipe.net/api/html/widgets/public/playout_cloudfront?token="
-      , "https://api.picturepipe.net/api/html/widgets/public/playout_cloudfront?token="
-      , "http://archives.crem-cnrs.fr/archives/items/"
-      , "http://www.theeuropeanlibrary.org/tel4/newspapers/issue/fullscreen/"
+  private static final Collection<String> URL_MATCHING_LIST = Arrays.asList(
+      //SoundCould
+      "http://soundcloud.com/",
+      "https://soundcloud.com/",
+      //Vimeo
+      "http://player.vimeo.com/video/",
+      "https://player.vimeo.com/video/",
+      "https://vimeo.com/",
+      "http://vimeo.com/",
+      //YouTube
+      "https://*.youtube.com/watch",
+      "https://*.youtube.com/v/",
+      "https://youtu.be/",
+      //SketchFab
+      "https://sketchfab.com/models/",
+      "https://skfb.ly/",
+      //Dismarc
+      "http://www.dismarc.org/player/",
+      "http://eusounds.ait.co.at/player/",
+      //Europeana
+      "http://www.ccma.cat/tv3/alacarta/programa/titol/video/",
+      "http://www.ina.fr/video/",
+      "http://www.ina.fr/*/video/",
+      "http://api.picturepipe.net/api/html/widgets/public/playout_cloudfront?token=",
+      "https://api.picturepipe.net/api/html/widgets/public/playout_cloudfront?token=",
+      "http://archives.crem-cnrs.fr/archives/items/",
+      "http://www.theeuropeanlibrary.org/tel4/newspapers/issue/fullscreen/",
+      //BritishLibrary
+      "http://sounds.bl.uk/embed/",
+      //EUScreen
+      "http://www.euscreen.eu/item.html"
   );
 
+  // Create patterns, update wildcards to proper format and add a wildcard at the end of each url
+  private static final Collection<Pattern> PATTERNS = URL_MATCHING_LIST.stream()
+      .map(prefix -> prefix.replace("*", ".*"))
+      .map(prefix -> prefix.concat(".*")).map(Pattern::compile)
+      .collect(Collectors.toList());
```

Comment:
This doesn't always work as a regex because `?` is  a reserved character, and it occurs in some of the URLS. So it should be escaped. And it is too easy to later add URLs with other escaped characters (e.g. `-` or `&`) without realizing it.

Perhaps a better sequence (using the `Pattern.quote()` method twice) is:
```
private static final Collection<Pattern> PATTERNS = URL_MATCHING_LIST.stream()
      .map(Pattern::quote)
      .map(prefix -> prefix.replace(Pattern.quote("*"), ".*"))
      .map(prefix -> prefix.concat(".*")).map(Pattern::compile)
      .collect(Collectors.toList());
```
And I think we could also use some comments above somewhere for future developers explaining how to maintain and add to this list.

Reply:
The quote example you wrote doesn't work, because the quote wraps a pattern with a prefix and suffix so that it's translated as literal but it doesn't add escape characters..
I see you point though, will try to find another way to solve this

Reply:
I can see that the solution works. I have another one to suggest that doesn't depend on having a constant string with all special characters in it:
1. split the url on occurrences of '*'
2. escape (i.e. quote) the parts
3. concatenate them with .* in between and after
Do you think this would work? Do you think this would be better?

Separately: I think we should we put a '\A' before to make sure that the string starts with the URL, instead of also matching some weird URL where it occurs somewhere in the middle... what do you think?

Reply:
It's a good alternative, I will try it out

Reply:
It seems to work well.